DoSomething/voting-app
# Validate phone numbers by country code with Laravel-Phone.

#### Changes

Fixes #472 by using [Laravel-Phone](https://github.com/Propaganistas/Laravel-Phone) to validate phone numbers by the current session's country code. Laravel-Phone is a wrapper of Google's libphonenumber.
#### Current Issues

~~This seems to work great for US users and 10-digit BR numbers, but fails when given an 11-digit BR number. I'm still trying to figure out why, but that's probably a blocker for merging this in.~~
Should be fixed after some revisions made below!
#### How should this be tested?

Automated tests should continue to pass. It'd also be worthwhile to pull down this branch and test with various phone number & country code combinations to see if you can find any unexpected behavior (in addition to the issue mentioned above).

One thing I'm concerned about is that libphonenumber tries to be very _smart_. For example, it marked my `(123) 456-5555` test number as invalid. This is obviously really nice if it works, but I'm afraid there might be edge cases that'd be hard to track down... does anyone have feelings on that?

/cc @angaither @DeeZone
